Update source rule in `Ranking_Criteria` (#6150)
ppy/osu-wiki: wiki/Ranking_Criteria/en.md

- **[Guest difficulty](/wiki/Beatmap/Guest_difficulty) creators, storyboarders, skinners and hitsounders must be added to the tags of a beatmap.** This is to give credit where credit is due and help others identify the main contributors of any given beatmap. Usernames containing single characters separated by spaces must have the spaces replaced with underscores.
- **[Primary metadata sources](/wiki/Beatmap/Primary_metadata_source) must be used as references for metadata.** Do not modify metadata from primary sources except to comply with formatting and standardisation rules below. If no sources are available, use what is most common and recognizable.
- **The artists of a song must be traceable to existing people.** If there is no existing person recorded to be the artist, then `Unknown Artist` is to be used. A song's sole artist cannot be a fictional character or program unless it is the artist's alias.

> - **You must use the Source field if the song comes from or is directly tied to another media such as a video game, movie, series, etc.** If the song was featured or tied to a media after it was released, the source field is optional. If a song has multiple potential sources, any options are valid. 

Reviewer: why is this reverted to "optional"? i think this is wrong?

Author: many older songs are obscure and hard to figure out if the media or song came first, and many songs that were tied to media later become prominently known for that media even if it wasn't the initial source. Disallowing it would make researching source much more difficult for little benefit. Similar opinions on this are expressed in the thread

>   - For remixes, arrangements, or covers based upon the original song, the source(s) of the original song would apply in the same way.

Reviewer: ```suggestion
  - For remixes, arrangements, or covers based upon the original song, the source of the original song should be used.
```
is this i'm the intended meaning?

>   - Songs specifically made for events such as contest entries or event theme songs may use the event name as a source.

Reviewer: why not just add the word "event" to the first line, alongside "video game"?

Author: will do

- **[Guest difficulty](/wiki/Beatmap/Guest_difficulty) creators, storyboarders, skinners and hitsounders must be added to the tags of a beatmap.** This is to give credit where credit is due and help others identify the main contributors of any given beatmap. Usernames containing single characters separated by spaces must have the spaces replaced with underscores.
- **[Primary metadata sources](/wiki/Beatmap/Primary_metadata_source) must be used as references for metadata.** Do not modify metadata from primary sources except to comply with formatting and standardisation rules below. If no sources are available, use what is most common and recognizable.
- **The artists of a song must be traceable to existing people.** If there is no existing person recorded to be the artist, then `Unknown Artist` is to be used. A song's sole artist cannot be a fictional character or program unless it is the artist's alias.

> - **You must use the Source field if the song comes from or is directly tied to another media such as a video game, movie, series, event, etc.** If the song was featured or tied to a media after it was released, the source field is optional. If a song has multiple potential sources, any options are valid. 

Reviewer: Am I the only one that finds it weird that we're adding an "optional" clause under "rules"?

Rewording since I still disagree with the version in this PR:

```suggestion
- **You must use the Source field if the song comes from or is directly tied to another media such as a video game, movie, series, event, etc.** This also applies if the song was featured or tied to a media after it was released, but is primarily known due to that source. If a song has multiple potential sources, any option is valid. 
```

Author: Oh, I understand better what your issue with the wording was now. I'll apply this change as it is fair and fits to what the rule was initially going for.
